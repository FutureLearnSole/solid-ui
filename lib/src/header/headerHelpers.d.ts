import { NamedNode } from 'rdflib';
declare type ThrottleOptions = {
    leading?: boolean;
    throttling?: boolean;
    trailing?: boolean;
};
export declare function getPod(): NamedNode;
export declare function throttle(func: Function, wait: number, options?: ThrottleOptions): (...args: any[]) => any;
export {};
//# sourceMappingURL=headerHelpers.d.ts.map