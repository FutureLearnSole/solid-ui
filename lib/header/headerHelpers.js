"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getPod = getPod;
exports.throttle = throttle;

var _rdflib = require("rdflib");

/*
    Copied from mashlib/src/global/metadata.ts
 */
function getPod() {
  // @@ TODO: This is given that mashlib runs on NSS - might need to change when we want it to run on other Pod servers
  return (0, _rdflib.sym)(document.location.origin).site();
}

function throttle(func, wait) {
  var options = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
  var context, args, result;
  var timeout = null;
  var previous = 0;

  var later = function later() {
    previous = !options.leading ? 0 : Date.now();
    timeout = null;
    result = func.apply(context, args);
    if (!timeout) context = args = null;
  };

  return function () {
    var now = Date.now();
    if (!previous && !options.leading) previous = now;
    var remaining = wait - (now - previous); // @ts-ignore

    context = this;
    args = arguments;

    if (remaining <= 0 || remaining > wait) {
      if (timeout) {
        clearTimeout(timeout);
        timeout = null;
      }

      previous = now;
      result = func.apply(context, args);
      if (!timeout) context = args = null;
    } else if (!timeout && options.trailing !== false) {
      timeout = setTimeout(later, remaining);
    }

    return result;
  };
}
//# sourceMappingURL=headerHelpers.js.map