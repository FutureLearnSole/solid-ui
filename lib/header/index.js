"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initHeader = initHeader;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _rdflib = require("rdflib");

var _authn = require("../authn/authn");

var _widgets = require("../widgets");

var _icon = require("./icon");

var _emptyProfile = require("./empty-profile");

var _headerHelpers = require("./headerHelpers");

/*
    This file was copied from mashlib/src/global/header.ts file. It is modified to
    work in solid-ui by adjusting where imported functions are found.
 */
function initHeader(_x, _x2) {
  return _initHeader.apply(this, arguments);
}

function _initHeader() {
  _initHeader = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(store, options) {
    var header, pod;
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            header = document.getElementById('PageHeader');

            if (header) {
              _context2.next = 3;
              break;
            }

            return _context2.abrupt("return");

          case 3:
            pod = (0, _headerHelpers.getPod)();

            _authn.solidAuthClient.trackSession(rebuildHeader(header, store, pod, options));

          case 5:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  }));
  return _initHeader.apply(this, arguments);
}

function rebuildHeader(header, store, pod, options) {
  return /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(session) {
      var user;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              user = session ? (0, _rdflib.sym)(session.webId) : null;
              header.innerHTML = '';
              _context.t0 = header;
              _context.next = 5;
              return createBanner(store, pod, user, options);

            case 5:
              _context.t1 = _context.sent;

              _context.t0.appendChild.call(_context.t0, _context.t1);

            case 7:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function (_x3) {
      return _ref.apply(this, arguments);
    };
  }();
}

function createBanner(_x4, _x5, _x6, _x7) {
  return _createBanner.apply(this, arguments);
}

function _createBanner() {
  _createBanner = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(store, pod, user, options) {
    var podLink, menu, banner;
    return _regenerator["default"].wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            podLink = document.createElement('a');
            podLink.href = pod.uri;
            podLink.classList.add('header-banner__link');
            podLink.innerHTML = _icon.icon;

            if (!user) {
              _context3.next = 10;
              break;
            }

            _context3.next = 7;
            return createUserMenu(store, user, options);

          case 7:
            _context3.t0 = _context3.sent;
            _context3.next = 11;
            break;

          case 10:
            _context3.t0 = createLoginSignUpButtons();

          case 11:
            menu = _context3.t0;
            banner = document.createElement('div');
            banner.classList.add('header-banner');
            banner.appendChild(podLink);
            banner.appendChild(menu);
            return _context3.abrupt("return", banner);

          case 17:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  }));
  return _createBanner.apply(this, arguments);
}

function createLoginSignUpButtons() {
  var profileLoginButtonPre = document.createElement('div');
  profileLoginButtonPre.classList.add('header-banner__login');
  profileLoginButtonPre.appendChild((0, _authn.loginStatusBox)(document, null, {}));
  return profileLoginButtonPre;
}

function createUserMenuButton(label, onClick) {
  var button = document.createElement('button');
  button.classList.add('header-user-menu__button');
  button.addEventListener('click', onClick);
  button.innerText = label;
  return button;
}

function createUserMenuLink(label, href) {
  var link = document.createElement('a');
  link.classList.add('header-user-menu__link');
  link.href = href;
  link.innerText = label;
  return link;
}

function createUserMenu(_x8, _x9, _x10) {
  return _createUserMenu.apply(this, arguments);
}

function _createUserMenu() {
  _createUserMenu = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(store, user, options) {
    var fetcher, loggedInMenuList, loggedInMenu, loggedInMenuTrigger, profileImg, loggedInMenuContainer, throttledMenuToggle, timer;
    return _regenerator["default"].wrap(function _callee4$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            fetcher = store.fetcher;

            if (!fetcher) {
              _context4.next = 4;
              break;
            }

            _context4.next = 4;
            return fetcher.load(user);

          case 4:
            loggedInMenuList = document.createElement('ul');
            loggedInMenuList.classList.add('header-user-menu__list');
            loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink('Show your profile', user.uri)));
            loggedInMenuList.appendChild(createUserMenuItem(createUserMenuButton('Log out', function () {
              return _authn.solidAuthClient.logout();
            })));
            loggedInMenu = document.createElement('nav');
            loggedInMenu.classList.add('header-user-menu__navigation-menu');
            loggedInMenu.setAttribute('aria-hidden', 'true');
            loggedInMenu.appendChild(loggedInMenuList);
            loggedInMenuTrigger = document.createElement('button');
            loggedInMenuTrigger.classList.add('header-user-menu__trigger');
            loggedInMenuTrigger.type = 'button';
            profileImg = getProfileImg(store, user);

            if (typeof profileImg === 'string') {
              loggedInMenuTrigger.innerHTML = profileImg;
            } else {
              loggedInMenuTrigger.appendChild(profileImg);
            }

            loggedInMenuContainer = document.createElement('div');
            loggedInMenuContainer.classList.add('header-banner__user-menu', 'header-user-menu');
            loggedInMenuContainer.appendChild(loggedInMenuTrigger);
            loggedInMenuContainer.appendChild(loggedInMenu);
            throttledMenuToggle = (0, _headerHelpers.throttle)(function (event) {
              return toggleMenu(event, loggedInMenuTrigger, loggedInMenu);
            }, 50);
            loggedInMenuTrigger.addEventListener('click', throttledMenuToggle);
            timer = setTimeout(function () {
              return null;
            }, 0);
            loggedInMenuContainer.addEventListener('mouseover', function (event) {
              clearTimeout(timer);
              throttledMenuToggle(event);
            });
            loggedInMenuContainer.addEventListener('mouseout', function (event) {
              timer = setTimeout(function () {
                return throttledMenuToggle(event);
              }, 200);
            });
            return _context4.abrupt("return", loggedInMenuContainer);

          case 27:
          case "end":
            return _context4.stop();
        }
      }
    }, _callee4);
  }));
  return _createUserMenu.apply(this, arguments);
}

function createUserMenuItem(child) {
  var menuProfileItem = document.createElement('li');
  menuProfileItem.classList.add('header-user-menu__list-item');
  menuProfileItem.appendChild(child);
  return menuProfileItem;
}

function getProfileImg(store, user) {
  var profileUrl = null;

  try {
    var _profileUrl = _widgets.widgets.findImage(user);

    if (!_profileUrl) {
      return _emptyProfile.emptyProfile;
    }
  } catch (_unused) {
    return _emptyProfile.emptyProfile;
  }

  var profileImage = document.createElement('div');
  profileImage.classList.add('header-user-menu__photo');
  profileImage.style.backgroundImage = "url(\"".concat(profileUrl, "\")");
  return profileImage;
}

function toggleMenu(event, trigger, menu) {
  var isExpanded = trigger.getAttribute('aria-expanded') === 'true';
  var expand = event.type === 'mouseover';
  var close = event.type === 'mouseout';

  if (isExpanded && expand || !isExpanded && close) {
    return;
  }

  trigger.setAttribute('aria-expanded', (!isExpanded).toString());
  menu.setAttribute('aria-hidden', isExpanded.toString());
}
//# sourceMappingURL=index.js.map