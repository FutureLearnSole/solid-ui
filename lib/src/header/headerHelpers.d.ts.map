{"version":3,"file":"headerHelpers.d.ts","sourceRoot":"","sources":["../../../src/header/headerHelpers.ts"],"names":[],"mappings":"AAGA,OAAO,EAAE,SAAS,EAAO,MAAM,QAAQ,CAAA;AAGvC,aAAK,eAAe,GAAG;IACnB,OAAO,CAAC,EAAE,OAAO,CAAC;IAClB,UAAU,CAAC,EAAE,OAAO,CAAC;IACrB,QAAQ,CAAC,EAAE,OAAO,CAAC;CACtB,CAAA;AAED,wBAAgB,MAAM,IAAK,SAAS,CAGnC;AAED,wBAAgB,QAAQ,CAAE,IAAI,EAAE,QAAQ,EAAE,IAAI,EAAE,MAAM,EAAE,OAAO,GAAE,eAAoB,GAAG,CAAC,GAAG,IAAI,EAAE,GAAG,EAAE,KAAK,GAAG,CAgC9G"}