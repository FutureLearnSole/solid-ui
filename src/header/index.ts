import { IndexedFormula, NamedNode, sym } from 'rdflib'
import { loginStatusBox, solidAuthClient } from '../authn/authn'
import { widgets } from '../widgets'
import { icon } from './icon'
import { log } from '../debug'
import { getClasses } from '../jss'
import { styleMap } from './styleMap'
// import { SolidSession } from '../../typings/solid-auth-client'
// import { emptyProfile } from './empty-profile'
// import { throttle } from '../helpers/throttle'
// import { getPod } from './metadata'
// Just for now..  ./metadata in mashlib
// 3 variables below including at_hash in Solid claim had to make camelcase...
interface SolidAuthorization {
  accessToken: string;
  clientId: string;
  idToken: string;
}

type Menu = {
  label: string,
  url: string
}
type HeaderOptions = {
  logo?: string,
  menuList?: Menu[]
}
interface SolidClaim {
  atHash: string;
  aud: string;
  azp: string;
  cnf: {
    jwk: string;
  };
  exp: number;
  iat: number;
  iss: string;
  jti: string;
  nonce: string;
  sub: string;
}
export interface SolidSession {
  authorization: SolidAuthorization;
  credentialType: string;
  idClaims: SolidClaim;
  idp: string;
  issuer: string;
  sessionKey: string;
  webId: string;
}

export function getPod (): NamedNode {
  // @@ TODO: This is given that mashlib runs on NSS - might need to change when we want it to run on other Pod servers
  return sym(document.location.origin).site()
}

export const emptyProfile = `
<svg xmlns="http://www.w3.org/2000/svg" width="26" height="26" viewBox="0 0 26 26" fill="none">
    <path fill-rule="evenodd" clip-rule="evenodd" d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z" fill="#D8D8D8" stroke="#8B8B8B"/>
    <mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="0" y="0" width="26" height="26">
        <path fill-rule="evenodd" clip-rule="evenodd" d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z" fill="white" stroke="white"/>
    </mask>
    <g mask="url(#mask0)">
        <path fill-rule="evenodd" clip-rule="evenodd" d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z" fill="#8B8B8B"/>
    </g>
</svg>`

type ThrottleOptions = {
  leading?: boolean;
  throttling?: boolean;
  trailing?: boolean;
}

export function throttle (func: Function, wait: number, options: ThrottleOptions = {}): (...args: any[]) => any {
  let context: any, args: any, result: any
  let timeout: any = null
  let previous = 0
  const later = function () {
    previous = !options.leading ? 0 : Date.now()
    timeout = null
    result = func.apply(context, args)
    if (!timeout) context = args = null
  }
  return function () {
    const now = Date.now()
    if (!previous && !options.leading) previous = now
    const remaining = wait - (now - previous)
    // @ts-ignore
    context = this
    args = arguments
    if (remaining <= 0 || remaining > wait) {
      if (timeout) {
        clearTimeout(timeout)
        timeout = null
      }
      previous = now
      result = func.apply(context, args)
      if (!timeout) context = args = null
    } else if (!timeout && options.trailing !== false) {
      timeout = setTimeout(later, remaining)
    }
    return result
  }
}

export async function initHeader (store: IndexedFormula, options: HeaderOptions) {
  const header = document.getElementById('PageHeader')
  if (!header) {
    return
  }
  const pod = getPod()
  log(pod)
  solidAuthClient.trackSession(rebuildHeader(header, store, pod, options))
}

function rebuildHeader (header: HTMLElement, store: IndexedFormula, pod: NamedNode, options: HeaderOptions) {
  return async (session: SolidSession | null) => {
    // const user = session ? sym(session.webId) : null
    // SAM - don't forget to put in the line above instead of the line below :)
    const user = sym('https://sharonstrats.inrupt.net/profile/card#me')
    log(user)
    header.innerHTML = ''
    header.appendChild(await createBanner(store, pod, user, options))
  }
}

async function createBanner (store: IndexedFormula, pod: NamedNode, user: NamedNode | null, options: HeaderOptions): Promise<HTMLElement> {
  const podLink = document.createElement('a')
  podLink.href = pod.uri
  podLink.classList.add('header-banner__link')
  podLink.innerHTML = icon

  const menu = user
    ? await createUserMenu(store, user, options)
    : createLoginSignUpButtons()

  const banner = document.createElement('div')
  banner.classList.add('header-banner')
  banner.appendChild(podLink)
  banner.appendChild(menu)

  return banner
}

function createLoginSignUpButtons () {
  const profileLoginButtonPre = document.createElement('div')
  profileLoginButtonPre.classList.add('header-banner__login')
  profileLoginButtonPre.appendChild(loginStatusBox(document, null, {}))
  return profileLoginButtonPre
}
async function openDashboardPane (outliner: any, pane: string): Promise<void> {
  outliner.showDashboard({
    pane
  })
}

function createUserMenuButton (label: string, onClick: EventListenerOrEventListenerObject): HTMLElement {
  const button = document.createElement('button')
  button.classList.add('header-user-menu__button')
  button.addEventListener('click', onClick)
  button.innerText = label
  return button
}

function getStyle (styleClass) {
  return styleMap[styleClass]
}

function addStyleClassToElement (element: any, styleClass: string) {
  const style = getStyle(styleClass)
  const { classes } = getClasses(document.head, { [styleClass]: style })
  element.classList.add(classes[styleClass])
}

function createUserMenuLink (label: string, href: string): HTMLElement {
  // if you have a uri then call that otherwise call on the button..
  // specify action in mashlib..  when you have onclick this function should be called
  // call property onclick if present it's a callback function, call upon that property
  // will outside of this function..
  const link = document.createElement('a')
  // link.classList.add('header-user-menu__link')
  const style = getStyle('header-user-menu__link')
  const { classes } = getClasses(document.head, { 'header-user-menu__link': style })
  link.classList.add(classes['header-user-menu__link'])
  link.href = href
  link.innerText = label
  return link
}

async function createUserMenu (store: IndexedFormula, user: NamedNode, options: HeaderOptions): Promise<HTMLElement> {
  const fetcher = (<any>store).fetcher
  if (fetcher) {
    // Making sure that Profile is loaded before building menu
    await fetcher.load(user)
  }
  // const outliner = getOutliner(document)
  // SAM Options: here is where I can take the list of options and add to loggedInMenuList...
  const loggedInMenuList = document.createElement('ul')
  // loggedInMenuList.classList.add('header-user-menu__list')
  addStyleClassToElement(loggedInMenuList, 'headerUserMenuList')
  loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink('Show your profile', user.uri)))
  if (options.menuList) {
    options.menuList.map(function (menuItem) {
      loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink(menuItem.label, menuItem.url)))
    })
  }
  /*
  const menuItems = await getMenuItems(outliner)
  menuItems.forEach(item => {
    loggedInMenuList.appendChild(createUserMenuItem(createUserMenuButton(item.label, () => openDashboardPane(outliner, item.tabName || item.paneName))))
  }) */
  loggedInMenuList.appendChild(createUserMenuItem(createUserMenuButton('Log out', () => solidAuthClient.logout())))

  const loggedInMenu = document.createElement('nav')
  loggedInMenu.classList.add('header-user-menu__navigation-menu')
  addStyleClassToElement(loggedInMenu, 'headerUserMenuNavigationMenu')
  loggedInMenu.setAttribute('aria-hidden', 'true')
  loggedInMenu.appendChild(loggedInMenuList)

  const loggedInMenuTrigger = document.createElement('button')
  loggedInMenuTrigger.classList.add('header-user-menu__trigger')
  loggedInMenuTrigger.type = 'button'
  /* const profileImg = getProfileImg(store, user)
  if (typeof profileImg === 'string') {
    loggedInMenuTrigger.innerHTML = profileImg
  } else {
    loggedInMenuTrigger.appendChild(profileImg)
  } */

  const loggedInMenuContainer = document.createElement('div')
  // loggedInMenuContainer.classList.add('header-banner__user-menu', 'header-user-menu')
  loggedInMenuContainer.appendChild(loggedInMenuTrigger)
  loggedInMenuContainer.appendChild(loggedInMenu)

  const throttledMenuToggle = throttle((event: Event) => toggleMenu(event, loggedInMenuTrigger, loggedInMenu), 50)
  loggedInMenuTrigger.addEventListener('click', throttledMenuToggle)
  let timer = setTimeout(() => null, 0)
  loggedInMenuContainer.addEventListener('mouseover', event => {
    clearTimeout(timer)
    throttledMenuToggle(event)
  })
  loggedInMenuContainer.addEventListener('mouseout', event => {
    timer = setTimeout(() => throttledMenuToggle(event), 200)
  })

  return loggedInMenuContainer
}

function createUserMenuItem (child: HTMLElement): HTMLElement {
  const menuProfileItem = document.createElement('li')
  menuProfileItem.classList.add('header-user-menu__list-item')
  menuProfileItem.appendChild(child)
  return menuProfileItem
}

async function getMenuItems (outliner: any): Promise<Array<{
  paneName: string;
  tabName?: string;
  label: string;
  icon: string;
}>> {
  return outliner.getDashboardItems()
}

function getProfileImg (store: IndexedFormula, user: NamedNode): string | HTMLElement {
  const profileUrl = widgets.findImage(user)
  if (!profileUrl) {
    return emptyProfile
  }
  const profileImage = document.createElement('div')
  profileImage.classList.add('header-user-menu__photo')
  profileImage.style.backgroundImage = `url("${profileUrl}")`
  return profileImage
}

function toggleMenu (event: Event, trigger: HTMLButtonElement, menu: HTMLElement): void {
  const isExpanded = trigger.getAttribute('aria-expanded') === 'true'
  const expand = event.type === 'mouseover'
  const close = event.type === 'mouseout'
  if ((isExpanded && expand) || (!isExpanded && close)) {
    return
  }
  trigger.setAttribute('aria-expanded', (!isExpanded).toString())
  menu.setAttribute('aria-hidden', isExpanded.toString())
}
