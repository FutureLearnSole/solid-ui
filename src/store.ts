// This module of solid-ui has a main quadstore for the app to use

const { LiveStore } = require('pane-registry')

const store = new LiveStore()

module.exports = store
