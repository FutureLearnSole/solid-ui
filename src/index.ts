/*
The MIT License (MIT)

Copyright (c) 2015-2016 Solid

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

If you would like to know more about the solid Solid project, please see
https://github.com/solid/solid
*/
'use strict'
/**
 * Provides a Solid client helper object (which exposes various static modules).
 * @module solidUi.js
 * @main solidUi.js
 */

/**
 * @class SolidUi
 * @static
 */

// REMOVE @ts-ignore as you migrate files to TypeScript
import * as rdf from 'rdflib' // pull in first avoid cross-refs
// @ts-ignore
import * as ns from './ns'
// @ts-ignore
import * as acl from './acl'
// @ts-ignore
import * as aclControl from './acl-control'
// @ts-ignore
import * as authn from './signin'
// @ts-ignore
import * as create from './create'
// @ts-ignore
import * as icons from './iconBase'
// @ts-ignore
import * as log from './log'
// @ts-ignore
import * as matrix from './matrix'
// @ts-ignore
import * as media from './media-capture'
// @ts-ignore
import * as messageArea from './messageArea'
// @ts-ignore
import { infiniteMessageArea } from './chat/infinite'
// @ts-ignore
import * as pad from './pad'
// @ts-ignore
import * as preferences from './preferences'
// @ts-ignore
import * as store from './store'
// @ts-ignore
import * as style from './style'
// @ts-ignore
import * as table from './table'
// @ts-ignore
import * as tabs from './tabs'
// @ts-ignore
import * as utils from './utils'
// @ts-ignore
import * as widgets from './widgets'

const UI = {
  ns,
  rdf,
  acl,
  aclControl,
  authn,
  create,
  dom: window.document, // Idea that UI.dom can be adapted in non-browser environments
  icons,
  log,
  matrix,
  media,
  messageArea,
  infiniteMessageArea,
  pad,
  preferences,
  store,
  style,
  table,
  tabs,
  utils,
  widgets
}

if (typeof window !== 'undefined') {
  ;(<any>window).UI = UI // Simpler access by non-node scripts
}
export default UI
