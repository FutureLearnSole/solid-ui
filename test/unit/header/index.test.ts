import { silenceDebugMessages } from '../../helpers/setup'
import { initHeader } from '../../../src/header'
import store from '../../../src/store'
// import { JSDOM } from 'jsdom'

silenceDebugMessages()
// const window = new JSDOM('<!DOCTYPE html><p>Hello world</p>').window
// const dom = window.document

describe('header', () => {
  const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
  it('is exposed on public API', async () => {
    expect(await initHeader(store, options)).toMatchSnapshot()
  })
})
