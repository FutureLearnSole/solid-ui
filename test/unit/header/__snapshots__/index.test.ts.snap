// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header is exposed on public API 1`] = `undefined`;
